lib/excon/socket.rb

```diff
@@ -61,9 +63,22 @@ def read(max_length = nil)
 
     def readline
       if @nonblock && RUBY_VERSION.to_f > 1.8_7
-        buffer = String.new
-        buffer << (read_nonblock(1) || raise(EOFError)) while buffer[-1] != "\n"
-        buffer
+        result = String.new
+        block = @read_buffer
+        @read_buffer = String.new
+
+        loop do
+          idx = block.index("\n")
+          if idx.nil?
+            result << block
+          else
+            add_to_read_buffer(block.slice!(idx+1, block.length))
```

Comment:
It looks like this is the only place where this method call is used. Is there a particular reason to have it separate, vs just having the related code inline here?

Reply:
We could inline it! I just preferred to name it for clarity.

Reply:
Sure. I think I lean toward inline to avoid the overhead of method calls. I know it's small, but want to keep inner loop streamlined where we can.